rust/actyx/swarm/harness/src/lib.rs

```diff
@@ -316,7 +317,21 @@ where
                 futures::pin_mut!(deadline);
                 while !peers.is_empty() {
                     let res = {
-                        let f = machine.select(|ev| m!(ev.borrow(), Event::Connected(p) => *p));
+                        let f = async {
+                            let x = machine.select(|ev| m!(ev.borrow(), Event::Connected(p) => *p));
+                            futures::pin_mut!(x);
+                            loop {
+                                let poll = futures::poll!(&mut x);
+                                match poll {
+                                    task::Poll::Ready(x) => return x,
+                                    task::Poll::Pending => {
+                                        tokio::task::yield_now().await;
```

Comment:
if this fixes it then the real fix is in `machine.select`

Reply:
This does not completely fix it unfortunately, at least with single core.